### PR TITLE
Display warning icon in Sync Settings cell when data syncing is disabled

### PR DESCRIPTION
**Pull request by ayoy**
Task/Issue URL: https://app.asana.com/0/0/1206213106333863/f

**Steps to test this PR**:
Verify disabling L1 feature flag according to [the spec](https://app.asana.com/0/0/1206215838630770/f).

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

**Review by SabrinaTardio — COMMENTED** [review]
We are not showing the alert or warning in the menu item here?

**Comment by ayoy** [comment]
We are supposed to show the warning in the settings cell.

File: DuckDuckGo/SettingsViewController.swift [hunk 1]
```diff
@@ -113,7 +113,7 @@ class SettingsViewController: UITableViewController {
     }
 
     private var shouldShowSyncCell: Bool {
-        return syncService.featureFlags.contains(.userInterface) || internalUserDecider.isInternalUser
```
Comment (ayoy) [inline DuckDuckGo/SettingsViewController.swift:116]: the `|| internalUserDecider.isInternalUser` is a remainder from the original feature flags PR when we were supposed to show the UI to internal users anyway, even if Sync top-level feature flag is disabled. We've replaced that on Apple platforms by setting top-level flag to "internal", so this code shouldn't end up here.

**Review by SabrinaTardio — APPROVED** [review]
Thanks for adding the alert!

**Pull request merged**